### PR TITLE
Removes the Detective's antag eligibility 

### PR DESCRIPTION
**Pull request by CydiaLamiales**
## About The Pull Request
Reverts https://github.com/BeeStation/BeeStation-Hornet/pull/6372
Removes the ability for the Detective to roll antagonist and gives him a mindshield upon spawning

## Why It's Good For The Game
I generally don't think someone with access to the brig, decurity and medical records, sechuds and a lethal sidearm plus security's honest trust should be antag eligible.
The original PR proposed that a traitorous detective would some sort of corrupt cop and that it would introduce paranoia. Did that happen? Not really, I'd say.
Security often times just sidelines the poor old Det, even when they're an antag trying to do a gimmick. And as for the 'corrupt cop' part... Most of the traitor detectives I've seen just go the "schut target in maints, frame it on another target' route. Or just unloading lead into HoS's back. I haven't seen much good roleplay coming out of antagonist detectives, just gaming.
Not to mention that it's a role that the security team is supposed to rely upon but... when they're not available it brings borderline OOC frustration. Plus the false arrests given by them further deepen the 'shitsec' mindset of non-sec players.
The proposed solution to that was HoS reviewing the evidence but then that's just... HoS doing Detective's job for them, isn't it.
So overall Traitor Detective I feel like doesn't really work in our current player and code environment.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>
Don't knows how to show a role not being eligible for an antag roll, sorry.
</details>

## Changelog
:cl:
tweak: the Detective can no longer roll antag
add: Gave the Detective the mindshield back
/:cl:

### CODE REVIEW

**Comment by CydiaLamiales** [comment]
> Going to need a detailed reason for the revert. Once that's in place, this should go back through the same voting process. The community voted for this change, they should be the ones to decide if it was a mistake.

Updated the reason.
Also the plethera of non-antag Dets going AFK near roundstart because they failed their antag roll that I've seen... I feel like it's on par with Paramed in the spot for the top antag roll role.

**Comment by Crossedfall** [comment]
> Also the plethera of non-antag Dets going AFK near roundstart because they failed their antag roll that I've seen

Ban them from antags for rolling and open the slot back up. That's an admin issue and isn't really a fair criticism levied against antag-detectives specifically imo.

File: code/game/gamemodes/dynamic/dynamic_rulesets_latejoin.dm [hunk 1]
```diff
@@ -170,7 +170,7 @@
 	name = "Heretic Smuggler"
 	antag_datum = /datum/antagonist/heretic
 	antag_flag = ROLE_HERETIC
-	protected_roles = list(JOB_NAME_SECURITYOFFICER, JOB_NAME_WARDEN, JOB_NAME_HEADOFPERSONNEL, JOB_NAME_DETECTIVE, JOB_NAME_HEADOFSECURITY, JOB_NAME_CAPTAIN)
+	protected_roles = list(JOB_NAME_SECURITYOFFICER, JOB_NAME_DETECTIVE, JOB_NAME_WARDEN, JOB_NAME_HEADOFPERSONNEL, JOB_NAME_DETECTIVE, JOB_NAME_HEADOFSECURITY, JOB_NAME_CAPTAIN)
```
Comment (lowwy) [inline code/game/gamemodes/dynamic/dynamic_rulesets_latejoin.dm]: duplicate JOB_NAME_DETECTIVE in this list

File: code/game/gamemodes/dynamic/dynamic_rulesets_roundstart.dm [hunk 2]
```diff
@@ -91,7 +91,7 @@
 	name = "Changelings"
 	antag_flag = ROLE_CHANGELING
 	antag_datum = /datum/antagonist/changeling
-	protected_roles = list(JOB_NAME_SECURITYOFFICER, JOB_NAME_WARDEN, JOB_NAME_DETECTIVE, JOB_NAME_HEADOFSECURITY, JOB_NAME_CAPTAIN)
+	protected_roles = list(JOB_NAME_SECURITYOFFICER, JOB_NAME_DETECTIVE, JOB_NAME_WARDEN, JOB_NAME_DETECTIVE, JOB_NAME_HEADOFSECURITY, JOB_NAME_CAPTAIN)
```
Comment (lowwy) [inline code/game/gamemodes/dynamic/dynamic_rulesets_roundstart.dm]: duplicate JOB_NAME_DETECTIVE in this list

File: code/game/gamemodes/dynamic/dynamic_rulesets_roundstart.dm [hunk 3]
```diff
@@ -128,7 +128,7 @@
 	name = "Heretics"
 	antag_flag = ROLE_HERETIC
 	antag_datum = /datum/antagonist/heretic
-	protected_roles = list(JOB_NAME_SECURITYOFFICER, JOB_NAME_WARDEN, JOB_NAME_DETECTIVE, JOB_NAME_HEADOFSECURITY, JOB_NAME_CAPTAIN)
+	protected_roles = list(JOB_NAME_SECURITYOFFICER, JOB_NAME_DETECTIVE, JOB_NAME_WARDEN, JOB_NAME_DETECTIVE, JOB_NAME_HEADOFSECURITY, JOB_NAME_CAPTAIN)
```
Comment (lowwy) [inline code/game/gamemodes/dynamic/dynamic_rulesets_roundstart.dm]: duplicate JOB_NAME_DETECTIVE in this list

File: code/game/gamemodes/dynamic/dynamic_rulesets_roundstart.dm [hunk 4]
```diff
@@ -638,7 +638,7 @@
 	name = "Incursion"
 	antag_flag = ROLE_INCURSION
 	antag_datum = /datum/antagonist/incursion
-	protected_roles = list(JOB_NAME_SECURITYOFFICER, JOB_NAME_WARDEN, JOB_NAME_DETECTIVE,JOB_NAME_CAPTAIN, JOB_NAME_HEADOFPERSONNEL, JOB_NAME_HEADOFSECURITY, JOB_NAME_CHIEFENGINEER, JOB_NAME_RESEARCHDIRECTOR, JOB_NAME_CHIEFMEDICALOFFICER)
+	protected_roles = list(JOB_NAME_SECURITYOFFICER, JOB_NAME_DETECTIVE, JOB_NAME_WARDEN, JOB_NAME_DETECTIVE,JOB_NAME_CAPTAIN, JOB_NAME_HEADOFPERSONNEL, JOB_NAME_HEADOFSECURITY, JOB_NAME_CHIEFENGINEER, JOB_NAME_RESEARCHDIRECTOR, JOB_NAME_CHIEFMEDICALOFFICER)
```
Comment (lowwy) [inline code/game/gamemodes/dynamic/dynamic_rulesets_roundstart.dm]: duplicate JOB_NAME_DETECTIVE in this list

File: code/game/gamemodes/dynamic/dynamic_rulesets_roundstart.dm [hunk 5]
```diff
@@ -690,7 +690,7 @@
 	name = "Assimilation"
 	antag_flag = ROLE_HIVE
 	antag_datum = /datum/antagonist/hivemind
-	protected_roles = list(JOB_NAME_SECURITYOFFICER, JOB_NAME_WARDEN, JOB_NAME_DETECTIVE,JOB_NAME_HEADOFSECURITY, JOB_NAME_CAPTAIN)
+	protected_roles = list(JOB_NAME_SECURITYOFFICER, JOB_NAME_DETECTIVE, JOB_NAME_WARDEN, JOB_NAME_DETECTIVE,JOB_NAME_HEADOFSECURITY, JOB_NAME_CAPTAIN)
```
Comment (lowwy) [inline code/game/gamemodes/dynamic/dynamic_rulesets_roundstart.dm]: duplicate JOB_NAME_DETECTIVE in this list

**Review by lowwy — COMMENTED** [review]
Some duplicate names in the list

File: code/game/gamemodes/dynamic/dynamic_rulesets_roundstart.dm (hunk 4 above)
Comment (CydiaLamiales) [reply code/game/gamemodes/dynamic/dynamic_rulesets_roundstart.dm]: Nvm I saw it

**Comment by Rukofamicom** [comment]
I'll give you my reasoning: Consistency and sensibility. 
The detective is a true and actual member of the security team and the crew should not be subject to the detective being antagonist-eligible for the same reason as the rest of security and heads of staff aren't - they're in a position of authority. If they aren't a full member of security and don't hold the same authority because they're some contracted outsider, they need to lose the extra security clearance/access just like the brig physician did. They need `ACCESS_SEC_DOORS`, `ACCESS_FORENSICS_LOCKERS`, and `ACCESS_WEAPONS`. No need for `ACCESS_BRIG` or `ACCESS_SEC_RECORDS`. 

> The original PR proposed that a traitorous detective would some sort of corrupt cop and that it would introduce paranoia. Did that happen? Not really, I'd say.

I gotta say I disagree with this. There is certainly some paranoia there, both in and out of security. 

Also gotta say that I don't have any qualm with the alleged balance implications behind the fact the detective spawns with a lethal-capable firearm. The .38 is a cool gun and isn't too strong even when used lethally. I had concerns about this initially but they have faded. 

The entirety of my issues with detective could be answered just as equally by removing their extra clearance and access.

**Comment by Fronsis** [comment]
> #7860
> 
> There seem to be no good reasons to actually remove detective's antag eligability, only reasons as to things that should be changed about the detective. They shouldn't have access to their gun on roundstart, they shouldn't have access to the security lathe gear room etc. All of these are easilly solvable

Yes, but at the same time i'm still supporting my feedback comment on that GH

As someone who plays sec on the daily, yeah.. having a detective antag to play against feels.. a little similar to when we had Heads as antagonists back on Golden, they just take cheap routes and it's.. just boring, but i'd say that's mostly because the players themselves don't try to do anything interesting or ''back-stabbing/hidden agent'' like, i think ever since it got merged i've only seen a ''planting fake evidence'' gimmick twice and the ocasional Det that frees some Syndicates from their cells.

In general they just sneakily print lethal ammo, dissapear from comms and go do their own thing, hindering sec during most of the round due to the lack of a forensic scanner or investigations(Compared to months ago, Security now lets the detective do their job and don't rush the locker to get the spare scanner unless absolutely necessary)

On top of that if you add the fact that they get access to Evidence Room.. 9/10 times they grab badly-secured contraband items for themselves(When there's no Warden or HoS most contraband goes there due to the lack of armory contraband locker) meaning any other antagonist that tries to plan a robbery or a way to get back has it denied due to the det securing the toys for themselves pretty much for free.

And as OP pointed out, it brings borderline OOC frustration because the way security plays, having a detective is important and in general most antag detectives decides to go gamer rather than do anything interesting, specially the mid-round detectives, so many times i've witnessed when we had a  solid case going, trying to finish tying clues with det with the forensic scanner and deduction.. when suddenly they stop answering, showing, etc because they're now focused on being antags.

#7860 Could help, but would need some fixes to let the actual, regular detective to help around properly, i've already gave my feedback on the other post in regards of that.

**Comment by Tsar-Salat** [comment]
Hm. I personally like removing the antag eligibility, but I think you should leave the mindshield out.

Detective is an obvious weak point in the Security structure, I think he should be targetable by conversion antags. It keeps the trust of him not being antag eligible, whilst keeping an in for ambitious antags.

It has the added benefit of not letting det print weapons and using the mindshield firing pin.

**Comment by CydiaLamiales** [comment]
Sure, I can remove the mindshield.

File: code/modules/jobs/job_types/detective.dm [hunk 6]
```diff
@@ -51,6 +51,8 @@
 		/obj/item/melee/classic_baton/police=1)
 	mask = /obj/item/clothing/mask/cigarette
 
+	implants = list(/obj/item/implant/mindshield)
+
```
Comment (CydiaLamiales) [inline code/modules/jobs/job_types/detective.dm:55]:
```suggestion
```

Comment (CydiaLamiales) [inline code/modules/jobs/job_types/detective.dm:55]:
```suggestion
```

**Comment by PowerfulBacon** [comment]
Ideally we want detective to be in a state like most antagonists, where the presence of one will disrupt the ability for investigations to be performed, but not completely eliminate them. Apart from having too much gear and access, which can be easilly solved, there is no reason detective shouldn't be an antagonists. A much prefered solution would be bringing back the PR that adjusted the detective to make them more suitable for antagonist roles.
While this may be an improvement over the current state, its a step back and adjusting them to play better as antag (changing the access they have, changing their weapon to not be granted roundstart, reducing some of their armour values) would be a significantly prefered solution.

Closing for the reasons above.

**Comment by CydiaLamiales** [comment]
Well, after some discussion in the discord, bacon re-opened the PR.

**Review by PowerfulBacon — APPROVED** [review]
I'm accepting this as a status quo until someone makes the detective role suck less

**Review by Rukofamicom — APPROVED** [review]
I thought for sure this would need to be changed in more than five places, but sure enough it doesn't.

**Comment by EvilDragonfiend** [comment]
Make sure if the reverts are correctly done because it excludes antag ineligibility from others.

**Comment by CydiaLamiales** [comment]
Just checked, he's in every protected_roles list.

**Comment by EvilDragonfiend** [comment]
I think you forgot the mindshield back

**Pull request merged**